modcp.php

```diff
@@ -3569,7 +3569,69 @@
 
 			if($post_ip_sql)
 			{
-				$query = $db->simple_select('posts', 'COUNT(pid) AS count', "$post_ip_sql AND visible >= -1");
+				$where_sql = '';
+				$unviewablefids = array();
+				$onlyusfids = array();
+
+				// Check group permissions if we can't view posts not started by us
+				$group_permissions = forum_permissions();
+				foreach($group_permissions as $fid => $forumpermissions)
+				{
+					if($forumpermissions['canview'] == 0 || $forumpermissions['canviewthreads'] == 0)
```

Comment:
Filtering by `get_unviewable_forums()` results would be better than this conditional since passwords are also checked there. `get_inactive_forums()` should also be used.